tsdb/compact.go

```diff
@@ -718,15 +718,28 @@ func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blockPopulator Bl
 }
 
 type BlockPopulator interface {
-	PopulateBlock(ctx context.Context, metrics *CompactorMetrics, logger log.Logger, chunkPool chunkenc.Pool, mergeFunc storage.VerticalChunkSeriesMergeFunc, blocks []BlockReader, meta *BlockMeta, indexw IndexWriter, chunkw ChunkWriter) error
+	PopulateBlock(ctx context.Context, metrics *CompactorMetrics, logger log.Logger, chunkPool chunkenc.Pool, mergeFunc storage.VerticalChunkSeriesMergeFunc, blocks []BlockReader, meta *BlockMeta, indexw IndexWriter, chunkw ChunkWriter, postingsFunc IndexReaderPostingsFunc) error
+}
+
+// IndexReaderPostingsFunc is a function to get a posting iterator from a given index reader.
+type IndexReaderPostingsFunc func(ctx context.Context, reader IndexReader) index.Postings
```

Comment:
I couldn't figure out from a few minutes' study of the diffs, why the contract says "a posting iterator" and the implementation is "all sorted postings".  These things are not similar.

Can you clarify, probably in this comment, whether `PopulateBlock` needs all postings and/or sorted postings.

Reply:
I see. For Posting iterator I initially meant `index.Postings`. But you are right, it should be a sorted postings iterator since it is requires for compaction. I can mention the contract to be `sorted`.